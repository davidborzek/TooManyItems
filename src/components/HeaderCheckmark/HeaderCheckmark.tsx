import { Ionicons } from '@expo/vector-icons';
import React from 'react';
import { TouchableOpacity } from 'react-native';

type Props = {
  disabled?: boolean;
  onPress?: () => void;
};

/**
 * Ein wiederverwendbarer Component welcher ein Anklickbares Checkmark bereitstellt.
 */
export default function HeaderCheckmark({ disabled, onPress }: Props) {
  return (
    <TouchableOpacity disabled={disabled} onPress={onPress}>
      <Ionicons
        name={'checkmark-outline'}
        size={24}
        style={{ opacity: disabled ? 0.3 : 1 }}
      />
    </TouchableOpacity>
  );
}
