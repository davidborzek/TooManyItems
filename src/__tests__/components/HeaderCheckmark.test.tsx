import HeaderCheckmark from '../../components/HeaderCheckmark/HeaderCheckmark';
import React from 'react';
import renderer from 'react-test-renderer';

it('HeaderCheckmark renders correctly', () => {
  const tree = renderer.create(<HeaderCheckmark />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('HeaderCheckmark renders correctly disabled', () => {
  const tree = renderer.create(<HeaderCheckmark disabled />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('HeaderCheckmark renders correctly with onPress', () => {
  const tree = renderer.create(<HeaderCheckmark onPress={() => ''} />).toJSON();
  expect(tree).toMatchSnapshot();
});
