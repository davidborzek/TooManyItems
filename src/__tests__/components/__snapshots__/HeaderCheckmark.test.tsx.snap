// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderCheckmark renders correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;

exports[`HeaderCheckmark renders correctly disabled 1`] = `
<View
  accessibilityState={
    Object {
      "disabled": true,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;

exports[`HeaderCheckmark renders correctly with onPress 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <Text />
</View>
`;
